test/test_replaygain.py

```diff
@@ -116,15 +151,85 @@ def test_cli_saves_track_gain(self):
                 mediafile.rg_track_gain, item.rg_track_gain, places=2)
 
     def test_cli_skips_calculated_tracks(self):
+        album_rg = self._add_album(1)
+        item_rg = album_rg.items()[0]
+
+        if self.has_r128_support:
+            album_r128 = self._add_album(1, ext="opus")
+            item_r128 = album_r128.items()[0]
+
         self.run_command(u'replaygain')
-        item = self.lib.items()[0]
-        peak = item.rg_track_peak
-        item.rg_track_gain = 0.0
+
+        item_rg.load()
+        self.assertIsNotNone(item_rg.rg_track_gain)
+        self.assertIsNotNone(item_rg.rg_track_peak)
+        self.assertIsNone(item_rg.r128_track_gain)
+
+        item_rg.rg_track_gain += 1.0
+        item_rg.rg_track_peak += 1.0
+        item_rg.store()
+        rg_track_gain = item_rg.rg_track_gain
+        rg_track_peak = item_rg.rg_track_peak
+
+        if self.has_r128_support:
+            item_r128.load()
+            self.assertIsNotNone(item_r128.r128_track_gain)
+            self.assertIsNone(item_r128.rg_track_gain)
+            self.assertIsNone(item_r128.rg_track_peak)
+
+            item_r128.r128_track_gain += 1.0
+            item_r128.store()
+            r128_track_gain = item_r128.r128_track_gain
+
         self.run_command(u'replaygain')
-        self.assertEqual(item.rg_track_gain, 0.0)
-        self.assertEqual(item.rg_track_peak, peak)
+
+        item_rg.load()
+        self.assertEqual(item_rg.rg_track_gain, rg_track_gain)
+        self.assertEqual(item_rg.rg_track_peak, rg_track_peak)
+
+        if self.has_r128_support:
+            item_r128.load()
+            self.assertEqual(item_r128.r128_track_gain, r128_track_gain)
+
+    def test_cli_does_not_skip_wrong_tag_type(self):
+        """Check that items that have tags of the wrong type won't be skipped.
+        """
+        if not self.has_r128_support:
+            # This test is a lot less interesting if the backend cannot write
+            # both tag types.
+            self.skipTest("r128 tags for opus not supported on backend {}"
+                          .format(self.backend))
+
+        album_rg = self._add_album(1)
+        item_rg = album_rg.items()[0]
+
+        album_r128 = self._add_album(1, ext="opus")
+        item_r128 = album_r128.items()[0]
+
+        item_rg.r128_track_gain = 0.0
+        item_rg.store()
+
+        item_r128.rg_track_gain = 0.0
+        item_r128.rg_track_peak = 42.0
+        item_r128.store()
+
+        self.run_command(u'replaygain')
+        item_rg.load()
+        item_r128.load()
+
+        self.assertIsNotNone(item_rg.rg_track_gain)
+        self.assertIsNotNone(item_rg.rg_track_peak)
+        # FIXME: Should the plugin null this field?
+        # self.assertIsNone(item_rg.r128_track_gain)
+
+        self.assertIsNotNone(item_r128.r128_track_gain)
+        # FIXME: Should the plugin null these fields?
```

Comment:
TBH, I have no idea. I don't really know how (precisely) these tags are specified, and how media players handle them in practice. There may be a point in nulling the "wrong" fields, in case they are really old and calculated for a different gain value for example. I'd advocate for leaving these comments in place, unless someone is sure of the correct approach.